components/formats-bsd/src/loci/formats/FileStitcher.java

```diff
@@ -144,6 +144,11 @@ public FileStitcher(IFormatReader r, boolean patternIds) {
    */
   public void setReaderClassList(ClassList<IFormatReader> classList) {
     this.classList = classList;
+    try {
+      reader.close();
```

Comment:
Is `reader.close(true)` (file handle-only close) enough to fix the original problem? That passes tests locally.

Reply:
Indeed, that's sufficient to fix the original issue while keeping the importer tests passing. Thanks, I'll mark this PR as ready for review and assign it to the next patch release